bot.js

```diff
@@ -266,7 +266,18 @@ function sendLongMessage(channel, text, length = 2000) {
 }
 
 function selectOne(array) {
-    return array[Math.round(Math.random() * (array.length - 1))];
+    // First, 'cook' array into a flat list of strings
+    // (allowing for the {option: '', weight: <n>} syntax)
+    const options = array.flatMap(option => {
```

Comment:
flatMap POGGERS

Reply:
Yes... Yes... Let the functional programming flow through you!